### ISSUE TITLE
Auto Load Images is no longer working

### ISSUE THREAD

**Issue by dbussert** [RocketChat/Rocket.Chat#763]
Did a 1-click deploy on Heroku for someone else, and the site is running fine, except no images are autoloading.  My user's `Auto Load Images` setting is default to true.  I went to an older version of Rocket.Chat on a VM I have an auto-loads work fine, so this leads me to believe this feature broken at some point recentyl.

**Comment by graywolf336** [comment]
Try something, go to the admin panel and change the "Allow invalid and Self-Signed SSL certificate's for link validation and previews" setting to allow and see if it works then:

![2015-09-11_12-04-15](https://cloud.githubusercontent.com/assets/850391/9821028/3fe13646-587d-11e5-9b9b-9008caaa5f53.png)

**Comment by dbussert** [comment]
@graywolf336 I deployed it right before that commit, so the setting wasn't there.  I blew that app away and did it again to get the latest commit.  I set it to true and no change, still does not auto-load.

**Comment by graywolf336** [comment]
That's as far as I can help, I haven't used it before with Heroku...sorry

**Comment by geekgonecrazy** [comment]
@dbussert alright when you say auto loading isn't working.  Is it showing something like this?

![image](https://cloud.githubusercontent.com/assets/51996/9821190/59b6b108-587e-11e5-878b-48f0b9d841f3.png)

Also is this only happening on uploads or link previews or both?

**Comment by dbussert** [comment]
more info, I took an older version of Rocket.Chat and upgraded it to latest and embeds work fine.  But deleting it and cloning off master embeds do not work.  I'll try cloning off some different commits and see if I can find more info.

**Comment by dbussert** [comment]
@geekgonecrazy I'm specifically testing auto-loading images, but its both off pasted urls and uploads

**Comment by geekgonecrazy** [comment]
@dbussert ok.  Does it show the click to load image?  If so does clicking it load? We need to make sure its getting that far.

**Comment by dbussert** [comment]
@geekgonecrazy neither of those, no action when pasting in a link

**Comment by geekgonecrazy** [comment]
Ok so this has something to do with the metadata collection step happening server side. I'll try a heroku deploy and see what I come up with.

If you want a bit more rapid troubleshooting.  Please feel free to join https://demo.rocket.chat and hit up the #general or #support channels.

**Comment by geekgonecrazy** [comment]
@dbussert just an update.  I am able to reproduce this.  Working on trying to fix it now.

**Comment by kakawait** [comment]
FYI using https://hub.docker.com/_/rocket.chat/ images loading work fines, using latest version with https://hub.docker.com/r/danieldent/meteor/ images loading does not work (even with self-signed certificat option).

**Comment by kakawait** [comment]
If it can help `git bisect` for *gif autoloading* test said me:

```
abf9acca9663d41aeeb30e94d0441dffe756b889 is the first bad commit
commit abf9acca9663d41aeeb30e94d0441dffe756b889
Author: Marcelo Schmidt <marcelo.schmidt@konecty.com>
Date:   Mon Aug 17 21:54:04 2015 -0300

    Settings: API_Embed; closes #225
```

**Comment by kakawait** [comment]
Flood! After testing issue is part of server not client. Be removing following line https://github.com/RocketChat/Rocket.Chat/blob/abf9acca9663d41aeeb30e94d0441dffe756b889/packages/rocketchat-oembed/server/server.coffee#L173 *gif autoloading test* works.

I'm not JS developer, even more Meteor so I can't help much. 

It strange because on client `RocketChat.settings.get 'API_Embed'` correctly returns `true` but on server it seems not (I don't know how to debug Meteor app, so I don't know if an error or just return `false`)

**Comment by rodrigok** [comment]
This issue probably was resolved by PR #849, if not, please reopen.

**Comment by flashguy22** [comment]
Unfortunately I can't get the inline image preview.
![bildschirmfoto 2015-10-23 um 10 03 04](https://cloud.githubusercontent.com/assets/1381617/10687418/4a68d4d2-796d-11e5-9c79-68ba85003d17.png)
We did empty the database. Is there any way to enable this again. restarting the server didn't work.
